units/crons.go

```diff
@@ -213,10 +215,63 @@ func PopulateEventAlertProcessing(env evergreen.Environment, parts int) amboy.Qu
 			return nil
 		}
 
-		ts := utility.RoundPartOfHour(parts).Format(TSFormat)
+		err = dispatchUnprocessedNotifications(ctx, queue, flags)
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "unable to dispatch unprocessed notifications",
+		}))
+		notifySettings := env.Settings().Notify
+		if err = notifySettings.Get(env); err != nil {
+			return errors.Wrap(err, "unable to get notify settings")
+		}
+		limit := notifySettings.EventProcessingLimit
+		if limit <= 0 {
+			limit = evergreen.DefaultEventProcessingLimit
+		}
+
+		events, err := event.FindUnprocessedEvents(-1)
+		if err != nil {
+			return errors.Wrap(err, "unable to find unprocessed events")
+		}
+		grip.Info(message.Fields{
+			"message": "unprocessed event count",
+			"pending": len(events),
+			"limit":   limit,
+			"source":  "events-processing",
+		})
+		if len(events) == 0 {
+			return nil
+		}
+
+		eventIDs := []string{}
+		for i, evt := range events {
+			eventIDs = append(eventIDs, evt.ID)
+			if (i+1)%limit == 0 || i+1 == len(events) {
+				err = queue.Put(ctx, NewEventNotifierJob(env, queue, sha256sum(eventIDs), eventIDs))
+				grip.Error(message.WrapError(err, message.Fields{
+					"message": "unable to queue event notifier job",
+				}))
+				eventIDs = []string{}
+			}
+		}
 
-		return errors.Wrap(queue.Put(ctx, NewEventMetaJob(env, queue, ts)), "failed to queue event-metajob")
+		return nil
 	}
+
+}
+
+// dispatchUnprocessedNotifications gets unprocessed notifications
+// leftover by previous runs and dispatches them
+func dispatchUnprocessedNotifications(ctx context.Context, q amboy.Queue, flags *evergreen.ServiceFlags) error {
```

Comment:
it seems like this function is only called once? I'm not sure that having a seperate function is needed, and I'm a little bit iffy about having functions other than the populate function in crons.go